src/R/Components/Impl/PackageManager/Implementation/RPackageManager.cs

```diff
@@ -165,46 +150,42 @@ internal class RPackageManager : IRPackageManager {
         private async Task<IReadOnlyList<RPackage>> GetPackagesAsync(Func<IRExpressionEvaluator, Task<REvaluationResult>> queryFunc) {
             // Fetching of installed and available packages is done in a
             // separate package query session to avoid freezing the REPL.
-            await CreatePackageQuerySessionAsync();
-
-            if (!_pkgQuerySession.IsHostRunning) {
-                throw new RPackageManagerException(Resources.PackageManager_EvalSessionNotAvailable);
-            }
-
-            try {
-                // Get the repos and libpaths from the REPL session and set them
-                // in the package query session
-                await EvalRepositoriesAsync(await DeparseRepositoriesAsync());
-                await EvalLibrariesAsync(await DeparseLibrariesAsync());
-
-                var result = await queryFunc(_pkgQuerySession);
-                CheckEvaluationResult(result);
+            using (var sessionToken = await _sessionPool.GetSession()) {
```

Comment:
`try/catch` should wrap `using (var sessionToken = await _sessionPool.GetSession())`, not vice versa